packages/expo-updates/ios/EXUpdates/Database/Migrations/EXUpdatesDatabaseMigration4To5.m

```diff
@@ -0,0 +1,56 @@
+//  Copyright © 2021 650 Industries. All rights reserved.
+
+#import <EXUpdates/EXUpdatesDatabaseMigration4To5.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+static NSString * const EXUpdatesDatabaseV4Filename = @"expo-v4.db";
+
+@implementation EXUpdatesDatabaseMigration4To5
+
+- (NSString *)filename
+{
+  return EXUpdatesDatabaseV4Filename;
+}
+
+- (BOOL)runMigrationOnDatabase:(struct sqlite3 *)db error:(NSError ** _Nullable)error
+{
+  // https://www.sqlite.org/lang_altertable.html#otheralter
+  if (sqlite3_exec(db, "PRAGMA foreign_keys=OFF;", NULL, NULL, NULL) != SQLITE_OK) return NO;
+  if (sqlite3_exec(db, "BEGIN;", NULL, NULL, NULL) != SQLITE_OK) return NO;
```

Comment:
Technically need to turn foreign key constraint checks back on. I think there are two ways to do this reasonably: split this function into two so that the outer function disables FK checks, runs the inner function, re-enables FK checks, and returns the result of the inner function; or use a labeled `goto`. Could set up a helper like this that disables/re-enables the FK checks around invoking the block.

```objc
return [self _execWithoutForeignKeyChecks:^ {
  ... exec calls
}];

Reply:
I think we are ok here, we have multiple checks in place for this already -- (1) if NO is returned then we assume the migration was unsuccessful and fallback to a destructive migration, so turning foreign keys on beforehand would be pointless, and (2) in any case (whether the migration is successful or not), we close the DB connection and open a new one before using the database, and we always turn on foreign keys immediately:

https://github.com/expo/expo/pull/12084/files#diff-8bde65cb2250b49c564e08711a0ad672bc80139e499630efa2ec4e218f18d267R135-R137

so in fact, we don't even technically need https://github.com/expo/expo/pull/12084/files#diff-c7623ef0a5018adcab7bc780490b7be75cc1913604c0f706053c7aaec38e6aabR41 for this reason.

This behavior is also tested in the unit test, which runs the full migration logic and then ensures the foreign key constraints are on.

Reply:
OK. (2) plus the tests are convincing to me. We just don't want the database session to be in an unexpected state somewhere else in the app's lifecycle.